File: app/Models/Student.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Student extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id', 'student_code', 'full_name',
        'email', 'phone', 'date_of_birth', 'address',
    ];

    protected $casts = [
        'date_of_birth' => 'date',
    ];

    //  Relations 
    public function user()
    {
        return $this->belongsTo(User::class);
    }

    /** Danh sach lop hoc vien dang/da tham gia */
    public function classes()
    {
        return $this->belongsToMany(SchoolClass::class, 'class_student', 'student_id', 'class_id')
                    ->withPivot('enrolled_at', 'status')
                    ->withTimestamps();
    }

    /** Bai tap da nop */
    public function assignmentSubmissions()
    {
        return $this->hasMany(AssignmentSubmission::class);
    }

}

File: mobile_app/lib/models/student.dart
class Student {
  const Student({
    required this.id,
    required this.studentCode,
    required this.fullName,
    this.email,
  });

  final int id;
  final String studentCode;
  final String fullName;
  final String? email;

  factory Student.fromJson(Map<String, dynamic> json) {
    return Student(
      id: json['id'] as int,
      studentCode: json['student_code']?.toString() ?? '',
      fullName: json['full_name']?.toString() ?? '',
      email: json['email']?.toString(),
    );
  }
}
